### PR TITLE
chore: reimplement fiber migrate

### PR DESCRIPTION
**Pull request by romange**
Before, we used a temporary fiber to pass along the migrated fiber to another proactor. Now, we simplify the transition by using the dispatcher fiber that also handles I/O loop. We switch to it and execute the delegation to another proactor.

Also, remove the old code and get rid of sccache that is not very reliable.

### CODE REVIEW

File: util/fibers/detail/fiber_interface.cc [hunk 1]
```diff
@@ -330,6 +316,35 @@ void FiberInterface::ExecuteOnFiberStack(PrintFn fn) {
   });
 }
 
+FiberInterface* FiberInterface::SwitchSetup() {
```
Comment (romange) [inline util/fibers/detail/fiber_interface.cc:319]: refactored out as a separate function and reused in both SwitchTo and SwitchToAndExecute

File: util/fibers/detail/fiber_interface.cc [hunk 2]
```diff
@@ -262,39 +261,26 @@ void FiberInterface::AttachThread() {
...
 ctx::fiber_context FiberInterface::SwitchTo() {
-  // We can not assert !wait_hook.is_linked() here, because for timeout operations,
-  // the fiber can be activated with the wait_hook still linked.
-  FiberInterface* prev = this;
-
-  auto& fb_initializer = FbInitializer();
-  std::swap(fb_initializer.active, prev);
-
-  uint64_t tsc = CycleClock::Now();
+  FiberInterface* prev = SwitchSetup();
```
Comment (romange) [inline util/fibers/detail/fiber_interface.cc:264]: no changes here.

File: util/fibers/detail/fiber_interface.h [hunk 3]
```diff
@@ -54,17 +54,19 @@ class FiberInterface {
  public:
   enum Type : uint8_t { MAIN, DISPATCH, WORKER };
 
+  // public hooks for intrusive data structures.
```
Comment (romange) [inline util/fibers/detail/fiber_interface.h:57]: just moved them here

File: util/fibers/proactor_base.h [hunk 4]
```diff
@@ -332,28 +332,23 @@ template <typename Func> auto ProactorBase::AwaitBrief(Func&& f) -> decltype(f()
...
 
-  // active->SuspendUntilWakeup();
   done.Wait();
 
   return std::move(mover).get();
 }
 
-// Runs possibly awating function 'f' safely in ContextThread and waits for it to finish,
-// If we are in the context thread already, runs 'f' directly, otherwise
+// Runs possibly awating function 'f' safely in proactor thread and waits for it to finish,
```
Comment (chakaz) [inline util/fibers/proactor_base.h:350]: if you're here: awating->awaiting :)

**Review by chakaz — APPROVED** [review]
LG although I think I grok ~50% of the change :)

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -61,15 +51,16 @@ jobs:
...
+
     - name: Build
       run: |
           cd ${{github.workspace}}/build
 
           ninja -k 5 base/all io/all strings/all util/all echo_server ping_iouring_server https_client_cli s3_demo
-          ${SCCACHE_PATH} --show-stats
 
     - name: Test
+      timeout-minutes: 2
```
[the remark this answers is not in the document]
Comment (romange) [reply .github/workflows/ci.yml:63]: all the tests in helio take 10s

File: util/fibers/detail/scheduler.cc [hunk 6]
```diff
@@ -666,36 +365,25 @@ void Scheduler::AttachCustomPolicy(DispatchPolicy* policy) {
...
     fiber.ExecuteOnFiberStack(fn);
   }
 }
 
+void Scheduler::SuspendAndExecuteOnDispatcher(std::function<void()> fn) {
+  CHECK(FiberActive() != dispatch_cntx_.get());
+
+  // All our dispatch policies add dispatcher to ready queue, hence it must be there.
+  CHECK(dispatch_cntx_->list_hook.is_linked());
+  ready_queue_.erase(FI_Queue::s_iterator_to(*dispatch_cntx_));
```
[the remark this answers is not in the document]
Comment (romange) [reply util/fibers/detail/scheduler.cc:383]: done

**Pull request merged**